buildpacks/python/python/function.go

```diff
@@ -66,23 +94,48 @@ func (f Function) Contribute(layer libcnb.Layer) (libcnb.Layer, error) {
 			layer.LaunchEnvironment.Default(envName, envValue)
 		}
 
+		if err := f.validate(functionModule, functionName); err != nil {
```

Comment:
Note that we are checking the function module and name that are set on this `Function` instance. This is a slight deviation from the `validation.go` implementation as it has logic to adjust the function module and name based on whether an env var overrides it (see [here](https://github.com/vmware-tanzu/function-buildpacks-for-knative/pull/137/files#diff-af4646e8568af1f9e2296d68edf77f7e7b4f3753c3612c4d0edcd866b05c0cb1L103) and [here](https://github.com/vmware-tanzu/function-buildpacks-for-knative/pull/137/files#diff-af4646e8568af1f9e2296d68edf77f7e7b4f3753c3612c4d0edcd866b05c0cb1L117)). That seems unnecessary, as at the end of the day, what needs to be checked is whatever function module/name is configured in this `Function` layer contributor.

Reply:
Side question: Is my understanding correct that the function class is supposed to be overridable at launch/run time via `MODULE_NAME` and `FUNCTION_NAME`? If that's the case, I'd argue that it still makes sense to check the function class that is currently known to this layer contributor, because it will have no way of telling the future and knowing what a user may set at launch.

Reply:
No longer part of this file. Will move commentary to `validation.go`